internal/provider/resource_vault_cluster.go

```diff
@@ -328,26 +351,45 @@ func resourceVaultClusterUpdate(ctx context.Context, d *schema.ResourceData, met
 		return diag.Errorf("unable to fetch Vault cluster (%s): %v", clusterID, err)
 	}
 
-	// Confirm public_endpoint or tier have changed.
-	if !(d.HasChange("tier") || d.HasChange("public_endpoint")) {
+	// Confirm at least on modifiable field has changed
+	if !d.HasChanges("tier", "public_endpoint", "paths_filter") {
 		return nil
 	}
 
 	if d.HasChange("tier") {
+		clusterToScale := cluster
+		destTier := vaultmodels.HashicorpCloudVault20201125Tier(strings.ToUpper(d.Get("tier").(string)))
 		if inPlusTier(string(cluster.Config.Tier)) {
-			return diag.Errorf("scaling Plus tier clusters is not yet allowed")
+			// Plus tier clusters scale as a group via the primary cluster.
+			// Therefore, the cluster may have already scaled due to another resource's update.
+			// First, check that the scaling operation is necessary.
+			if cluster.Config.Tier == destTier {
+				clusterToScale = nil
+			} else {
+				printPlusScalingWarningMsg()
+				primaryLink := getPrimaryLinkIfAny(d)
+				if primaryLink != "" {
+					// If the cluster is a secondary, issue the actual API request to the primary.
```

Comment:
The API changes that would enable removing this special case are not currently planned, so that kind of comment wouldn't be accurate. Okay to leave as is, or would you like more explanation in the comments?